tree/ntupleutil/v7/src/RNTupleInspector.cxx

```diff
@@ -181,6 +181,24 @@ ROOT::Experimental::RNTupleInspector::Create(std::string_view ntupleName, std::s
    return inspector;
 }
 
+std::string ROOT::Experimental::RNTupleInspector::GetCompressionSettingsAsString() const
+{
+   int algorithm = fCompressionSettings / 100;
+   int level = fCompressionSettings - (algorithm * 100);
+
+   std::string algorithmName;
+   switch (algorithm) {
+   case 1: algorithmName = "zlib"; break;
+   case 2: algorithmName = "LZMA"; break;
+   case 3: algorithmName = "old compression algorithm"; break;
+   case 4: algorithmName = "lz4"; break;
+   case 5: algorithmName = "zstd"; break;
+   default: algorithmName = "undefined compression algorithm";
```

Comment:
Now that I give this a second look, how can it happen? Isn't maybe worth it throwing an exception in case we get a file with an unrecognised compression settings? Can this be problematic elsewhere in code @jblomer ?

Reply:
It may happen for files from the future that come with a not yet known compression algorithm. I think that's no issue for the inspector, the current approach looks fine to me.